google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/metrics/BigtableExporterUtils.java

```diff
@@ -0,0 +1,255 @@
+import java.util.Collection;
+import java.util.List;
+import java.util.Set;
+import java.util.UUID;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
+/** Utils to convert OpenTelemetry types to Cloud Monitoring API types. */
+class BigtableExporterUtils {
+
+  private static final Logger logger = Logger.getLogger(BigtableExporterUtils.class.getName());
+
+  static String getDefaultTaskValue() {
+    // Something like '<pid>@<hostname>'
+    final String jvmName = ManagementFactory.getRuntimeMXBean().getName();
+    // If not the expected format then generate a random number.
+    if (jvmName.indexOf('@') < 1) {
+      String hostname = "localhost";
+      try {
+        hostname = InetAddress.getLocalHost().getHostName();
+      } catch (UnknownHostException e) {
+        logger.log(Level.INFO, "Unable to get the hostname.", e);
+      }
+      // Generate a random number and use the same format "random_number@hostname".
+      return "java-" + UUID.randomUUID() + "@" + hostname;
+    }
+    return "java-" + UUID.randomUUID() + jvmName;
+  }
+
+  private static final Set<AttributeKey<String>> PROMOTED_RESOURCE_LABELS =
+      ImmutableSet.of(PROJECT_ID, INSTANCE_ID, TABLE_ID, CLUSTER_ID, ZONE_ID);
+
+  static List<TimeSeries> convertCollectionToListOfTimeSeries(
+      Collection<MetricData> collection, String taskId, MonitoredResource monitoredResource) {
+    List<TimeSeries> allTimeSeries = new ArrayList<>();
+
+    for (MetricData metricData : collection) {
+      if (!metricData.getInstrumentationScopeInfo().getName().equals("bigtable.googleapis.com")) {
+        continue;
+      }
+
+      List<TimeSeries> timeSeries =
+          metricData.getData().getPoints().stream()
+              .map(
+                  pointData ->
+                      convertPointToTimeSeries(metricData, pointData, taskId, monitoredResource))
+              .collect(Collectors.toList());
+      allTimeSeries.addAll(timeSeries);
+    }
+
+    return allTimeSeries;
+  }
+
+  private static TimeSeries convertPointToTimeSeries(
+      MetricData metricData,
+      PointData pointData,
+      String taskId,
+      MonitoredResource monitoredResource) {
+    Attributes attributes = pointData.getAttributes();
+    MonitoredResource.Builder monitoredResourceBuilder = monitoredResource.toBuilder();
+    ImmutableMap.Builder<String, String> metricLabels = new ImmutableMap.Builder<>();
+
+    // Populated monitored resource schema
+    for (AttributeKey<?> attributeKey : PROMOTED_RESOURCE_LABELS) {
+      monitoredResourceBuilder.putLabels(
+          attributeKey.getKey(), String.valueOf(attributes.get(attributeKey)));
+    }
+
+    // Populate the rest of the metric labels
+    attributes.forEach(
+        (key, value) -> {
+          if (!PROMOTED_RESOURCE_LABELS.contains(key) && value != null) {
+            metricLabels.put(key.getKey(), String.valueOf(value));
+          }
+        });
```

Comment:
these 2 are basically doing the same thing, but using 2 different loop styles. The only practical difference where bucket to put the entry into.

I think it would be clearer to do this:

```
for(String key : attributes) {
  String value = String.valueOf(attributes.get(key));
  if (PROMOTED_RESOURCE_LABELS.contains(key)) {
    resourceBuilder.putLabel(key, value);
  } else {
    metricLabels.put(...)
}
```

Reply:
One more thing: why bother with intermedia metricLabels Map? why not just add it directly to the Metric?